library/network_connections.py

```diff
@@ -222,31 +244,29 @@ def link_infos(cls, refresh=False):
         return linkinfos
 
     @classmethod
-    def link_info_find(cls, refresh=False, mac=None, ifname=None):
-        if mac is not None:
+    def link_info_find(cls, mac=None, ifname=None):
```

Comment:
You should remove searching link by mac address because that will cause ambiguity. This function should accept interface name only and all user of this function should adopt to it. Otherwise, you will always raise false alarm somewhere.

Reply:
I adopted your suggestion and updated the code, thanks!